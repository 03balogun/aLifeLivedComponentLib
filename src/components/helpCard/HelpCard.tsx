import React from 'react';
import { Box } from '../../themes/Box';
import { Text } from '../../themes/Text';

export const HelpCard: React.FC<{ title: string; subText: string }> = ({
  title,
  subText,
}) => {
  return (
    <Box margin="m" borderBottomWidth={1} paddingBottom="s">
      <Text variant="h3">{title}</Text>
      <Text variant="caption" marginTop="m">
        {subText}
      </Text>
    </Box>
  );
};
