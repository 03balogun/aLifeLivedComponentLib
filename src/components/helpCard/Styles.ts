import { StyleSheet, TextStyle, ViewStyle } from 'react-native';

interface Styles {
  container: ViewStyle;
  textHeader: TextStyle;
  subText: TextStyle;
}

const styles = StyleSheet.create<Styles>({
  container: {
    margin: 10,
    paddingTop: 10,
    paddingBottom: 10,
    borderBottomWidth: 1,
  },

  textHeader: {
    fontWeight: 'bold',
    color: 'grey',
    fontSize: 10,
  },

  subText: {
    marginTop: 10,
    color: 'grey',
  },
});

export default styles;
