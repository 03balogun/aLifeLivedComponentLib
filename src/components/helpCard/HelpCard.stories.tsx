import React from 'react';
import { storiesOf } from '@storybook/react-native';

// Component
import { HelpCard } from './HelpCard';

storiesOf('Help Card', module).add('with text', () => (
  <HelpCard title="Help me!" subText="This is the sub text" />
));
