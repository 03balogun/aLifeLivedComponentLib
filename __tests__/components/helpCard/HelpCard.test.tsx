import React from 'react';
import { render } from '@testing-library/react-native';

// Component
import { HelpCard } from '../../../src/components/helpCard/HelpCard';

const props = {
  title: 'Test title',
  subText: 'Test sub text',
};

const setup = () => {
  // Screen
  const utils = render(
    <HelpCard title={props.title} subText={props.subText} />
  );

  return {
    ...utils,
  };
};

describe('Help card component test ', () => {
  test('Should render help card correctly', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  test('Should render title and sub text', () => {
    const { getByText } = setup();
    expect(getByText(props.title)).not.toBeNull();
    expect(getByText(props.subText)).not.toBeNull();
  });
});
