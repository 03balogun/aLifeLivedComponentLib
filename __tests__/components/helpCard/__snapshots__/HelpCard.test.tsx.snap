// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help card component test  Should render help card correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "display": "flex",
        "margin": 16,
        "paddingBottom": 8,
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "display": "flex",
          "fontSize": 16,
          "lineHeight": 24,
        },
      ]
    }
  >
    Test title
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#54575a",
          "display": "flex",
          "fontSize": 14,
          "lineHeight": 18,
          "marginTop": 16,
        },
      ]
    }
  >
    Test sub text
  </Text>
</View>
`;
